src/utils.js

```diff
@@ -55,6 +55,38 @@ export const deepClone = source => {
   return extend(target, source);
 };
 
+export const softOverwriteState = (oldState, newState) => {
```

Comment:
What is the difference between `softOverwriteState` and `extend`? It seems like the two functions are identical.